### ISSUE TITLE
New create  but  block  is red

### ISSUE THREAD

**Issue by ErNisha4800** [LLK/scratch-gui#9535]
![block red](https://github.com/scratchfoundation/scratch-gui/assets/149057719/27b6a8a7-922b-4d4f-be10-2085ab120331)

there are 3 file add
- scratch-blocks/blocks_vertical/motion.js
               // motion.js
Blockly.Blocks['motion_move100steps'] = {
  /**
   * Block to move 100 steps.
   * @this Blockly.Block
   */
  init: function() {
    this.jsonInit({
      "message0": Blockly.Msg.MOTION_MOVE100STEPS,
      "category": Blockly.Categories.motion,
      "extensions": ["colours_motion", "shape_statement"]
    });
  }
};
- scratch-blocks/msg/messages.js

Blockly.Msg.MOTION_MOVE100STEPS = 'move 100 steps';

- scratch-gui/src/lib/make-toolbox-xml.js

<block type="motion_move100steps" />

<block type="motion_movesteps">
  <value name="STEPS">
    <shadow type="math_number">
      <field name="NUM">2</field>
    </shadow>
  </value>
</block>

**Comment by PaperMarioFan-2022** [comment]
I think you misunderstood something. That is the red hat block and it was removed, according to the Scratch Wiki: https://en.scratch-wiki.info/wiki/Undefined_Hat_Block

I doubt the Scratch Team will even add it back, unless for AFD, which seems like it fits well for an april fools joke.

**Comment by CST1229** [comment]
> I think you misunderstood something. That is the red hat block and it was removed, according to the Scratch Wiki: https://en.scratch-wiki.info/wiki/Undefined_Hat_Block

I think they mean that they're trying to add a block but it becomes the red hat block, not that they're suggesting the ST to add it.

To the original poster: This likely means you didn't correctly compile and link your modified ersion of scratch-blocks.

**Comment by PaperMarioFan-2022** [comment]
> > I think you misunderstood something. That is the red hat block and it was removed, according to the Scratch Wiki: https://en.scratch-wiki.info/wiki/Undefined_Hat_Block
> 
> I think they mean that they're trying to add a block but it becomes the red hat block, not that they're suggesting the ST to add it.
> 
> To the original poster: This likely means you didn't correctly compile and link your modified ersion of scratch-blocks.

Oh, okay. Thanks for clarification.

**Comment by ErNisha4800** [comment]
Yes, I want to do the same , but it doesn't happen , can you help

**Comment by PaperMarioFan-2022** [comment]
> Yes, I want to do the same , but it doesn't happen , can you help

Maybe the ST haven't implemented adding random blocks, yet? They added custom blocks that you could use instead.

**Comment by rubiidev18alt** [comment]
I'm guessing this is what the block was supposed to look like.
![scratchblocks (1)](https://github.com/scratchfoundation/scratch-gui/assets/164731988/54310d71-312a-438e-ba2e-6a5c6a6901d5)

**Comment by JonathanTilahun** [comment]
What is the setup to get this bug. I am not getting it.

**Comment by PaperMarioFan-2022** [comment]
> What is the setup to get this bug. I am not getting it.

The OP didn't clarify a set-up for this.

**Comment by dumorando** [comment]
see the guide at https://mrcomputer1.github.io/scratch-modding/tutorial/getting-started/prerequisites
theres a good tutorial for adding blocks :)
